WinFormUserInterface.cs

```diff
@@ -0,0 +1,137 @@
+            foreach (string subdirectory in subdirectoryEntries)
+            {
+                DirectoryInfo di = new DirectoryInfo(subdirectory);
+                TreeNode tds = td.Nodes.Add(di.Name);
+                tds.StateImageIndex = 0;
+                tds.Tag = di.FullName;
+                LoadFiles(subdirectory, tds);
+                LoadSubDirectories(subdirectory, tds);
+            }
+        }
+
+
+
+        private void LoadFiles(string dir, TreeNode td)
+        {
+            string[] Files = Directory.GetFiles(dir, "*.*");
+
+            // Loop through them to see files  
+            foreach (string file in Files)
+            {
+                FileInfo fi = new FileInfo(file);
+                TreeNode tds = td.Nodes.Add(fi.Name);
+                tds.Tag = fi.FullName;
+                tds.StateImageIndex = 1;
+            }
+        }
+
+        public IEnumerable<string> SafeEnumerateFiles(string path, string searchPattern = "*.*", SearchOption searchOption = SearchOption.TopDirectoryOnly)
+        {
+            var dirs = new Stack<string>();
+            dirs.Push(path);
+
+            while (dirs.Count > 0)
+            {
+                string currentDirPath = dirs.Pop();
+                if (searchOption == SearchOption.AllDirectories)
+                {
+                    try
+                    {
+                        string[] subDirs = Directory.GetDirectories(currentDirPath);
+                        foreach (string subDirPath in subDirs)
+                        {
+                            dirs.Push(subDirPath);
+                        }
+                    }
+                    catch (UnauthorizedAccessException)
+                    {
+                        continue;
+                    }
+                    catch (DirectoryNotFoundException)
+                    {
+                        continue;
+                    }
+                }
+
+                string[] files = null;
+                try
+                {
+                    files = Directory.GetFiles(currentDirPath, searchPattern);
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    continue;
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    continue;
+                }
+
+                foreach (string filePath in files)
+                {
+                    yield return filePath;
+                }
+            }
+        }
+
+        public string FormatBytes(long bytes)
+        {
+            string[] Suffix = { "B", "KB", "MB", "GB", "TB" };
+            int i;
+            double dblSByte = bytes;
+            for (i = 0; i < Suffix.Length && bytes >= 1024; i++, bytes /= 1024)
+            {
+                dblSByte = bytes / 1024.0;
+            }
+
+            return String.Format("{0:0.##} {1}", dblSByte, Suffix[i]);
```

Comment:
Не модно сейчас. Сейчас модна строковая интерполяция.
```suggestion
            return $"{dblSByte:0.##} {Suffix[i]}";
```

Reply:
Компилятор тогда сам сможет принять решение как лучше оптимизировать написанное Вами.